plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -59,8 +58,8 @@
     private final Optional<String> viewMaterializationProject;
     private final Optional<String> viewMaterializationDataset;
     private final boolean caseInsensitiveNameMatching;
-    private final Cache<String, String> remoteDatasetNames;
-    private final Cache<String, Map<String, String>> remoteTableNames;
+    private final Cache<String, Optional<String>> remoteDatasetNames;
+    private final Cache<TableId, Optional<TableId>> remoteTableNames;
```

Comment:
We talked about it but actually it it looks like values can be `Optional<String>`. We are getting `getTableName` alway anyway. Is the there a benefit to store `TableId` as a value?

Reply:
Hmmm, seems you are correct. The projectId is always lowercase and the dataset is already resolved to the remote name so those values aren't useful.

I'll update.